apps/web/pages/week/index.page.tsx

```diff
@@ -53,24 +56,29 @@ const WeekPage = () => {
       <div className="mb-4 flex items-end justify-between">
         {myProjectsMembersData.length > 0 && (
           <Listbox
-            value={myProjectsMembersData.find((user) => user.id === selectedUserId) ?? myProjectsMembersData[0]}
+            value={myProjectsMembersData.find((user) => user.id === selectedUserId) ?? { id: 'all', name: 'All Users' }}
             getLabel={(user) => <UserLabel name={user.name ?? user.id} image={user.image ?? undefined} />}
             getKey={(user) => user.id}
             onChange={(user) => handleUserChange(user.id)}
-            options={myProjectsMembersData}
+            options={[{ id: 'all', name: 'All Users' }, ...myProjectsMembersData]}
           />
         )}
         <div className="flex grow justify-center">
           <WeekSelector value={day} onChange={handleWeekChange} />
         </div>
       </div>
-      {!weekGridData && fetching && <div className="loading loading-spinner" />}
+      {!weekGridData && fetching && (
+        <div className="flex h-full items-center justify-center">
+          <div className="loading loading-spinner size-10" />
+        </div>
+      )}
       {weekGridData?.projects && (
         <WeekGrid
           tableData={weekGridData.projects}
           startDate={startDate}
           endDate={endDate}
           isDataOutdated={isDataOutdated}
+          currentUserId={currentUserId}
```

Comment:
I would rename:
all props that can also have the value "all" I would name: selectedUser
all props (deeper in rendering) that must not contain anything but real db ids I would rename from currentUserId into userId, because the term currentUserId was from the time when we only had one current user and this user was not able to edit other users workhours.